Wrappers/Python/test/test_out_in_place.py

```diff
@@ -313,3 +315,132 @@ def test_proximal_out(self):
+        except (InPlaceError, NotImplementedError):
+            print("in_place_test test not implemented for  " + processor.__class__.__name__)
+
+    def test_out(self):
+        
+        for geom in self.geometry_test_list:
+            for data_array in self.data_arrays:
+                data=geom.allocate(None)
+                try:
+                    data.fill(data_array)
+                except:
+                    data.fill(np.repeat(data_array[:,None, :], repeats=2, axis=1))
+
+                # Add new Processors here 
+                # Works when out is same or different size to the data
+                processor_list = [
+                    TransmissionAbsorptionConverter(min_intensity=0.01),
+                    AbsorptionTransmissionConverter(),
+                    RingRemover(),
+                    Slicer(roi={'horizontal':(None,None,None),'angle':(None,None,None)}), 
+                    Slicer(roi={'horizontal':(1,3,2),'angle':(None,4,2)}), 
+                    Binner(roi={'horizontal':(None,None,None),'angle':(None,None,None)}),
+                    Binner(roi={'horizontal':(1,None,2),'angle':(None,4,2)}),
+                    Padder(pad_width=0),
+                    Padder(pad_width=1),
+                    Normaliser(flat_field=data.get_slice(angle=0).as_array()*1, dark_field=data.get_slice(angle=0).as_array()*1e-5),
+                    MaskGenerator.median(threshold_factor=3, window=7),
+                    Masker.median(mask=data.copy())
+                    ]                
+                
+                for processor in processor_list:
+                    self.out_tests(processor, data)
+   
+    def test_centre_of_rotation_out(self):
+        for geom in self.geometry_test_list[0:2]:
+            for data_array in self.data_arrays:
+                data=geom.allocate(None)
+                try:
+                    data.fill(data_array)
+                except:
+                    data.fill(np.repeat(data_array[:,None, :], repeats=2, axis=1))
+                processor = CentreOfRotationCorrector.xcorrelation(ang_tol=180)
+                processor.set_input(data)
+                out1 = processor.get_output()
+                # out_test fails because the processor only updates geometry, I think this is expected behaviour
+                # self.out_test(result, processor, data)
+                # test geometry instead
+                input = data.copy()
+                out2 = 0*(out1.copy())
+                try:
+                    processor.set_input(input)
+                    out3 = processor.get_output(out=out2)
+                    
+                    numpy.testing.assert_array_equal(out1.geometry.config.system.rotation_axis.position, out2.geometry.config.system.rotation_axis.position, err_msg= "Calculation failed using processor.set_input(data), processor.get_output(out=out) where func is  " + processor.__class__.__name__+ ".")
```

Comment:
You probably want `numpy.testing.assert_allclose` as you're comparing floating point values.

Reply:
Update now